sonic.go

```diff
@@ -159,13 +159,25 @@ func (cfg *frozenConfig) Valid(data []byte) bool {
 // Opts are the compile options, for example, "option.WithCompileRecursiveDepth" is
 // a compile option to set the depth of recursive compile for the nested struct type.
 func Pretouch(vt reflect.Type, opts ...option.CompileOption) error {
-    if err := encoder.Pretouch(vt, opts...); err != nil {
-        return err
-    } else if err = decoder.Pretouch(vt, opts...); err != nil {
-        return err
-    } else {
-        return nil
-    }
+	if err := encoder.Pretouch(vt, opts...); err != nil {
+	    return err
+	} 
+	if err := decoder.Pretouch(vt, opts...); err != nil {
+		return err
+	}
+	// to pretouch the corresponding pointer type as well
+	if vt.Kind() == reflect.Ptr {
+		vt = vt.Elem()
+	} else {
+		vt = reflect.PtrTo(vt)
+	}
+	if err := encoder.Pretouch(vt, opts...); err != nil {
```

Comment:
possible repeat code ？

Reply:
- Pretouch() will pretouch concrete type and pointer type both for one call now